### PR TITLE
Update format.conf for vehicle parts

### PR DESCRIPTION
**Pull request by sethsimon**
This updates `format.conf` to include any legal syntax for the `vehicle_part` type. It's [CR] because although I did my best to thoroughly test it, because I know nothing about JSON and even less about `lint`, I may have made a mistake somewhere.

### CODE REVIEW

**Comment by mugling** [comment]
Did you check out the `README` for the linter?

**Comment by mugling** [comment]
Needs extension of test cases in `examples` covering the new fields

**Comment by sethsimon** [comment]
I read it, it was extremely helpful. Because the build failed, I'll have to run it locally to see what context in `blaze_oerride.json` is causing trouble. And I'd be happy to update the examples.

**Comment by mugling** [comment]
Presuming you have no JSON errors it should give you the exact name of the failing context from which you can derive a rule

**Comment by mugling** [comment]
> I read it, it was extremely helpful.

Glad to here it. That you were able to extend the ruleset without external help is encouraging.

**Comment by sethsimon** [comment]
The reason the build failed is that one of my rule changes affected `data/mods/blazemod/blaze_override.json`, which is in the `json_whitelist`. I changed `^vehicle_part:breaks_into:@:charges` to `^vehicle_part:breaks_into:@:charges=NOWRAP` because
`{ "item": "cable", "charges": [ 3, 6 ] }` is a lot less ugly than:
`{`
`        "item": "cable", "charges": [`
`          3,`
`          6`
`        ]`
`      }`

**Comment by sethsimon** [comment]
This is ready.

File: tools/format/examples/vehicle_part.json [hunk 1]
```diff
@@ -14,14 +14,15 @@
     "breaks_into": [
       { "item": "steel_lump", "count": [ 3, 6 ] },
       { "item": "steel_chunk", "count": [ 3, 6 ] },
-      { "item": "scrap", "count": [ 3, 6 ] }
+      { "item": "scrap", "min": 3, "max": 6 }
```
Comment (mugling) [inline tools/format/examples/vehicle_part.json]: If `min`, `max` and `count` are interchangeable we should perhaps aim to support only one variant?

Comment (mugling) [inline tools/format/examples/vehicle_part.json]: It looks like `min` and `max` aren't actually red

Comment (sethsimon) [inline tools/format/examples/vehicle_part.json]: > If `min`, `max` and `count` are interchangeable we should perhaps aim to support only one variant?

Yeah, that sounds good.

> It looks like `min` and `max` aren't actually read.

Good catch. It looks like you beat me to it, thanks.

Comment (mugling) [inline tools/format/examples/vehicle_part.json]: No worries. Now that you have the hang of I'd encourage you to submit more linter rules so we can extend coverage to most of the core JSON

**Pull request merged**